modules/wallet/addresses.go

```diff
@@ -16,8 +16,13 @@ func (w *Wallet) timelockedCoinAddress(unlockHeight consensus.BlockHeight) (coin
 	spendConditions = consensus.SpendConditions{
 		TimeLock:      unlockHeight,
 		NumSignatures: 1,
-		PublicKeys:    []crypto.PublicKey{pk},
+		PublicKeys: []consensus.SiaPublicKey{
+			consensus.SiaPublicKey{
+				Algorithm: consensus.ED25519Identifier,
+			},
+		},
 	}
+	copy(spendConditions.PublicKeys[0].Key, pk[:])
```

Comment:
`crypto` wouldn't return a `SiaPublicKey`, the `wallet` would take care of that. Anyway, as it stands it's definitely broken, but the fix is easy.